geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java

```diff
@@ -688,7 +688,7 @@ public void basicProcess() {
       if (isAckRequired()) {
         ack();
       }
-      if (!GemFireCacheImpl.getExisting("Applying TXCommitMessage").isClient()) {
```

Comment:
Looking at the code above in the try block; the cache could be null and needs to be handled...
Line#607
      final Cache cache = dm.getExistingCache();
      if (cache == null) {
        addProcessingException(new CacheClosedException());
        // return ... this cache is closed so we can't do anything.
        return;
      }

Reply:
Eric and I also noticed that code and walked through it. That logic is faulty.
getExistingCache is designed to NEVER return null. So that call to addProcessingException will never be called.
I think this old code should have instead called getInstance instead of getExisting. Then it could get null back. Eric and I decided to file a separate bug for this.